pkg/machines/components/vcpuModal.jsx

```diff
@@ -0,0 +1,188 @@
+            state.sockets = divs[divs.length - 1];
+        }
+        divs = dividers(state.max / state.sockets);
+        if (divs.indexOf(this.state.cores) === -1) {
+            state.cores = divs[divs.length - 1];
+        }
+        state.threads = state.max / ( state.cores * state.sockets );
+        this.setState(state);
+    }
+
+    onCountSelect (e) {
+        let value = parseInt(e.target.value);
+        if (isNaN(value) || value <= 0) {
+            value = 1;
+        } else {
+            if (value > this.state.max) {
+                value = this.state.max;
+            }
+        }
+        this.setState({ count: parseInt(value) });
+    }
+
+    onSocketChange (value) {
+        const state = { sockets: this.state.sockets, cores: this.state.cores };
+        state.sockets = parseInt(value);
+        let divs = dividers(this.state.max / parseInt(value));
+        if (divs.indexOf(this.state.cores) === -1) {
+            state.cores = divs[divs.length - 1];
+        }
+        state.threads = (this.state.max/(state.sockets*state.cores));
+        this.setState(state);
+    }
+
+    onThreadsChange (value) {
+        const state = { sockets: this.state.sockets, threads: this.state.threads };
+        state.threads = parseInt(value);
+        let divs = dividers(this.state.max / state.threads);
+        if (divs.indexOf(state.sockets) === -1) {
+            state.sockets = divs[divs.length - 1];
+        }
+        state.cores = (this.state.max/(state.sockets*state.threads));
+
+        this.setState(state);
+    }
+
+    onCoresChange (value) {
+        const state = { sockets: this.state.sockets, threads: this.state.threads };
+        state.cores = parseInt(value);
+        let divs = dividers(this.state.max / state.cores);
+        if (divs.indexOf(state.sockets) === -1) {
+            state.sockets = divs[divs.length - 1];
+        }
+        state.threads = (this.state.max/(state.sockets*state.cores));
+        this.setState(state);
+    }
+
+
+    render () {
+        return (<div className="modal-body">
+            <table className="vcpu-detail-modal-table">
+                <tr>
+                    <td>
+                        <table className='form-table-ct'>
+                            <InfoRecord
+                                descr={_("vCPU Count")}
+                                tooltip={_("Specify fewer than the maximum number of virtual CPUs should be enabled.")}
+                                value={<input id="machines-vcpu-count-field" type="number" className="form-control" value={this.state.count} onChange={this.onCountSelect} />}
+                            />
+                            <InfoRecord
+                                descr={_("vCPU Maximum")}
+                                tooltip={_("Maximum number of virtual CPUs allocated for the guest OS, which must be between 1 and the maximum supported by the hypervisor.")}
```

Comment:
This tooltip could be made obsolete by the UI checking what the maximum number of CPUs is on the hypervisor, and then limiting the spinner to that maximum number and giving you an error if you input too many.